include/picongpu/particles/atomicPhysics/SetToAtomicGroundStateForChargeState.hpp

```diff
@@ -52,16 +52,12 @@ namespace picongpu
 
                     PMACC_DEVICE_ASSERT_MSG(numberBoundElectrons >= 0, "Number of bound electrons must be >= 0");
 
-                    using Particle = T_Particle;
-
-                    particle[boundElectrons_] = numberBoundElectrons;
```

Comment:
Is there no need to set `particle[boundElectrons_]` anymore?

Reply:
is done in the ChargeState manipulator itself, line 57 ChargeState.hpp, to keep atomicPhysics and charge state separate.

Reply:
@BrianMarre Since @pordyna reported that this PR is the reason for a deadlock in the simulation this question is showing up again.
In an old PR bound electrons were manipulated in the ionization implementation. Later it was removed by calling `SetToAtomicGroundStateForChargeState` 

https://github.com/ComputationalRadiationPhysics/picongpu/pull/4134/files#diff-f536e1a1139f8a19bc17a44e02c62eb2a728cb75b9373b23bcf061c0986966a3R322


Now you wrote that bound electrons will be set in `ChargeState.hpp` but the problem is that the ionization implementations does not call this functor. Therefore IMO `boundElectrons_` will never be updated.

Reply:
I opened issue #4611 to document the debugging

Reply:
You are correct, I missed that side effect, sorry :(